From: Bastian Krause <bst@pengutronix.de>
Date: Thu, 22 Oct 2020 10:16:49 +0200
Subject: [PATCH] WebProcess: InjectedBundle: guard setGenericCueAPIEnabled
 usage

The declaration in WebCore::Settings is guarded by..

  #if ENABLE(VIDEO) .. #endif

Also guard its use here.

This allows building with ENABLE_VIDEO=OFF.

Signed-off-by: Bastian Krause <bst@pengutronix.de>
---
 Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp b/Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp
index 61326f2e5731..d7776997aa08 100644
--- a/Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp
+++ b/Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp
@@ -236,12 +236,14 @@ void InjectedBundle::overrideBoolPreferenceForTestRunner(WebPageGroupProxy* page
         RuntimeEnabledFeatures::sharedFeatures().setWebRTCMDNSICECandidatesEnabled(enabled);
 #endif
 
+#if ENABLE(VIDEO)
     if (preference == "WebKitGenericCueAPIEnabled") {
         WebPreferencesStore::overrideBoolValueForKey(WebPreferencesKey::genericCueAPIEnabledKey(), enabled);
         for (auto* page : pages)
             page->settings().setGenericCueAPIEnabled(enabled);
         return;
     }
+#endif
 
 #if ENABLE(GPU_PROCESS)
     if (preference == "WebKitUseGPUProcessForMedia" || preference == "WebKitCaptureAudioInGPUProcessEnabledKey") {
